### DISCUSSION TITLE
Retrieve exact image match using public api while specifying all option parameters?

### DISCUSSION THREAD

**Discussion by bmm-brobbins** [dicebear/dicebear#216] [Q&A] [answered]
Hello, great project.  I have a little program that generates images locally, saves the options present in each image, and constructs a url for your public api at https://avatars.dicebear.com/api/ with all the options as parameters.  I've noticed that the images returned from the public api aren't an exact match with the option parameters as specified in the url.  It seems like some parameters are respected, others not.

My questions: 1) Is this expected behavior, or am I doing something wrong?  2) Either way, is there an alternate method to accomplish this?

Examples:
1)  Notice that the glasses don't appear.  They are present in my locally generated image (attached)
https://avatars.dicebear.com/api/micah/.svg?eyeColor=calm&base=standard&mouth=surprised&eyebrows=eyelashesDown&hair=full&eyes=round&nose=curve&ears=attached&shirt=open&glasses=square&baseColor=topaz&earringColor=sky&eyeShadowColor=azure&eyebrowColor=black&facialHairColor=topaz&glassesColor=mellow&hairColor=lavender&mouthColor=black&shirtColor=black

2)  Notice that the earring doesn't appear.  It is present in my locally generated image (attached).
https://avatars.dicebear.com/api/micah/.svg?eyeColor=calm&base=standard&mouth=frown&eyebrows=down&hair=turban&eyes=eyesShadow&nose=tound&ears=detached&shirt=open&earrings=hoop&baseColor=topaz&earringColor=topaz&eyeShadowColor=azure&eyebrowColor=black&facialHairColor=topaz&glassesColor=sky&hairColor=topaz&mouthColor=black&shirtColor=azure

I've also tried: 3) Using the open-peeps style instead, with similar results (i.e. some parameters take, others not); 4)  Constructing each parameter with '[]' notation as I saw somewhere in your docs; 5) Different settings for :seed.svg.  In the end I just removed the seed entirely in case that was getting in the way.

Thank you very much for any info.

![BARC2](https://user-images.githubusercontent.com/50039334/168223321-ffaf282c-8c57-4389-8f56-7963cb89f443.png)
![BARC3](https://user-images.githubusercontent.com/50039334/168223350-df7544d1-c48d-4a91-bb2d-9979763a96dc.png)

**Comment by FlorianKoerner** [comment, marked as answer]
Hi @bmm-brobbins 😃 

if you want to get the same result guaranteed, make sure you use the same seed. In your example URL, the seed is an empty string. If you don't specify one locally, it is a random value, so you can get different results.

See example 1:
https://codesandbox.io/s/dicebear-215-u6vtxy?file=/index.ts

Alternatively, you really need to set all options. This also includes the probability. Some elements are only displayed with a probability calculated on the seed.

See example 2:
https://codesandbox.io/s/dicebear-215-u6vtxy?file=/index.ts

**Reply by bmm-brobbins** [reply]
Terrific!  Your second example, setting the probabilities to 100, seems to do the trick.  Thanks a lot for the quick response and solution.

Btw, I'm only generating the avatar options locally because I couldn't figure out a way to capture the options associated with an otherwise randomly generated avatar.  Is there a way to do that using one of the public methods (API, CLI, etc.)?

**Reply by FlorianKoerner** [reply]
Sorry for the late reply. Currently the library does not return the used options. But it is a good idea. It has been requested several times.

https://github.com/dicebear/dicebear/issues/102#issuecomment-796879934

I will try to include this functionality in the upcoming versions.
